it/it-tests/src/test/java/it/issue/IssueTagsTest.java

```diff
@@ -0,0 +1,153 @@
+  @ClassRule
+  public static UserRule userRule = UserRule.from(orchestrator);
+
+  private String organizationKey;
+
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    enableOrganizationsSupport();
+  }
+
+  @Before
+  public void setUp() throws Exception {
+    organizationKey = ItUtils.newOrganizationKey();
+    newAdminWsClient(orchestrator).organizations().create(CreateWsRequest.builder()
+      .setKey(organizationKey)
+      .setName(organizationKey)
+      .build());
+  }
+
+  @Test
+  public void getTags() throws Exception {
+    restoreProfile(orchestrator, IssueTagsTest.class.getResource("/issue/one-issue-per-line-profile.xml"), organizationKey);
+    String projectKey = newProjectKey();
+    ItUtils.newAdminWsClient(orchestrator).projects().create(
+      CreateRequest.builder()
+        .setKey(projectKey)
+        .setOrganization(organizationKey)
+        .setName(randomAlphabetic(10))
+        .setVisibility("private")
+        .build());
+    analyzeProject(projectKey);
+
+    String issue = newAdminWsClient(orchestrator).issues().search(new SearchWsRequest()).getIssues(0).getKey();
+    newAdminWsClient(orchestrator).issues().setTags(issue, "bla", "blubb");
+
+    String[] publicTags = {"bad-practice", "convention", "pitfall"};
+    String[] privateTags = {"bad-practice", "bla", "blubb", "convention", "pitfall"};
+
+    // anonymous must not see custom tags of private project
+    {
+      assertTags(null, null, organizationKey, publicTags);
+      assertTags(null, null, null, publicTags);
+    }
+
+    // stranger must not see custom tags of private project
+    {
+      String stranger = randomAlphabetic(10).toLowerCase();
+      String strangerPassword = randomAlphabetic(8);
+      userRule.createUser(stranger, strangerPassword);
+      assertTags(stranger, strangerPassword, organizationKey, publicTags);
+      assertTags(stranger, strangerPassword, null, publicTags);
+    }
+
+    // member with user permission must be able to see custom tags of private project, if he provides the organization parameter
+    {
+      String member = randomAlphabetic(10).toLowerCase();
+      String memberPassword = randomAlphabetic(8);
+      userRule.createUser(member, memberPassword);
+      makeMember(member);
+      grantUserPermission(projectKey, member);
+      assertTags(member, memberPassword, organizationKey, privateTags);
+      assertTags(member, memberPassword, null, publicTags);
+    }
+  }
+
+  private void makeMember(String member) {
+    newAdminWsClient(orchestrator).organizations().addMember(organizationKey, member);
+  }
+
+  private void grantUserPermission(String projectKey, String member) {
+    newAdminWsClient(orchestrator).permissions().addUser(
+      new AddUserWsRequest()
+        .setLogin(member)
+        .setPermission("user")
+        .setProjectKey(projectKey));
+  }
+
+  private void assertTags(String user, String password, @Nullable String organization, String... expectedTags) {
```

Comment:
I agree that the current signature is not perfect - at very least I should have make `user` and `password` nullable. About the suggestion to provide an extra method I am a bit unsure, as it might lead in code duplication or higher complexity. Anyways I would prefer not to do the change at the moment, if that is ok for you.